cmd/rpcdaemon/commands/eth_receipts.go

```diff
@@ -676,7 +676,17 @@ func (api *APIImpl) GetTransactionReceipt(ctx context.Context, txnHash common.Ha
 		return nil, fmt.Errorf("block has less receipts than expected: %d <= %d, block: %d", len(receipts), int(txnIndex), blockNum)
 	}
 
-	return marshalReceipt(receipts[txnIndex], block.Transactions()[txnIndex], cc, block.HeaderNoCopy(), txnHash, true), nil
+	effectiveGasPricePercentage, err := api.getEffectiveGasPricePercentage(tx, txn.Hash())
+	if err != nil {
+		return nil, err
+	}
+
+	fields := marshalReceipt(receipts[txnIndex], block.Transactions()[txnIndex], cc, block.HeaderNoCopy(), txnHash, true)
+
+	// zkevm - override effectiveGasPrice to consider effectiveGasPricePercentage
+	fields["effectiveGasPrice"] = core.CalculateEffectiveGas(txn.GetPrice(), effectiveGasPricePercentage)
```

Comment:
can we extract all that in a single function (in `eth_receipts_zkevm.go`) that this all code will be like that

```
return addEffectiveGasPercentage(marshalReceipt(receipts[txnIndex], block.Transactions()[txnIndex], cc, block.HeaderNoCopy(), txnHash, true)), nil
```

Reply:
yeah good shout